Kentico.Kontent.Delivery.Extensions.Autofac.DependencyInjection/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Kentico.Kontent.Delivery.Abstractions;
+using Kentico.Kontent.Delivery.Caching;
+using Kentico.Kontent.Delivery.Caching.Extensions;
+using Kentico.Kontent.Delivery.Configuration;
+using Kentico.Kontent.Delivery.Helpers;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using System;
+
+namespace Kentico.Kontent.Delivery.Extensions.Autofac.DependencyInjection.Extensions
+{
+    /// <summary>
+    /// A class which contains extension methods on <see cref="IServiceCollection"/> for registering an <see cref="IDeliveryClient"/> instance.
+    /// </summary>
+    public static class ServiceCollectionExtensions
```

Comment:
can we somehow make it easier to register the required named services such as `ITypeProvider`, and the others like inlince content resolvers? Perhaps one method that would accept all the services and whose parameters would be null by default and the only required param would be `name` and that would register all the services under the given name?

Reply:
after an internal discussion: we'll leave that up to the consumer. we just need to document it well.
if needed, we can add something later.